microsoft/testsuites/power/common.py

```diff
@@ -32,33 +32,66 @@ def is_distro_supported(node: RemoteNode) -> None:
         )
 
 
-def verify_hibernation(node: RemoteNode, log: Logger) -> None:
+def verify_hibernation(
+    node: RemoteNode,
+    log: Logger,
+    index: int = 1,
+    ignore_call_trace: bool = False,
+) -> None:
+    hibernation_setup_tool = node.tools[HibernationSetup]
+    startstop = node.features[StartStop]
+
     node_nic = node.nics
     lower_nics_before_hibernation = node_nic.get_lower_nics()
     upper_nics_before_hibernation = node_nic.get_nic_names()
-    hibernation_setup_tool = node.tools[HibernationSetup]
     entry_before_hibernation = hibernation_setup_tool.check_entry()
     exit_before_hibernation = hibernation_setup_tool.check_exit()
     received_before_hibernation = hibernation_setup_tool.check_received()
     uevent_before_hibernation = hibernation_setup_tool.check_uevent()
-    startstop = node.features[StartStop]
+
+    # only set up hibernation setup tool for the first time
     hibernation_setup_tool.start()
-    startstop.stop(state=features.StopState.Hibernate)
+    sleep(300)
+    try:
+        startstop.stop(state=features.StopState.Hibernate)
+    except Exception as identifier:
+        try:
+            dmesg = node.tools[Dmesg]
+            content = dmesg.get_output(force_run=True)
+            log.debug(content)
+        except Exception as identifier_dmesg:
+            log.debug(
+                f"identifier_dmesg exception is {identifier_dmesg.__class__.__name__}: "
+                f"{identifier_dmesg}"
+            )
+        log.debug(f"exception is {identifier.__class__.__name__}: {identifier}")
+        raise identifier
+
     is_ready = True
     timeout = 900
     timer = create_timer()
     while timeout > timer.elapsed(False):
-        is_ready, _ = wait_tcp_port_ready(
-            node.connection_info[constants.ENVIRONMENTS_NODES_REMOTE_ADDRESS],
-            node.connection_info[constants.ENVIRONMENTS_NODES_REMOTE_PORT],
-            log=log,
-            timeout=10,
-        )
-        if not is_ready:
+        if "VM deallocated" == startstop.get_status():
```

Comment:
The status should be abstracted in Azure as an enumeration, not use the string directly.